conf/init.sql

```diff
@@ -31,11 +31,12 @@ create table if not exists glutz.config
 
 create table if not exists glutz.devices
 (
+    config_id           bigint not null,
     project_id          text not null,
     asset_id            integer not null,
     device_id           text not null,
     location_id         text not null,
-    primary key(project_id, device_id, location_id)
+    primary key(project_id, device_id)
```

Comment:
You haveto add config_id to the primary key. It should be possible, that the smae device id can exists in the same project but created nased on different config ids.

=> primary key(config_id, project_id, device_id)

Also it's a good database table design, to put all primary key fields at the top of the table in the order of the primary key definition. So the table's columns should start with config_id, project_id, device_id and then comes the other columns.